xc/device.ml

```diff
@@ -2319,7 +2319,12 @@ module Backend = struct
     module Monitor = struct
       module Epoll = Core.Linux_ext.Epoll
       module Flags = Core.Linux_ext.Epoll.Flags
-      let create () = (Core.Or_error.ok_exn Epoll.create) ~num_file_descrs:1001 ~max_ready_events:1
+      let num_file_descrs =
+        match Core.Unix.RLimit.((get num_file_descriptors).cur) with
+        | Core.Unix.RLimit.Limit v -> Int64.to_int v
+        | Core.Unix.RLimit.Infinity -> Core.Int.max_value
```

Comment:
num_file_descrs will be passed to core Epoll.create, so using Int.max_value might end up allocating an enormous amount of memory inside Table:
https://github.com/janestreet/core/blob/master/src/linux_ext.ml#L885
```
  let create ~num_file_descrs ~max_ready_events =
    if max_ready_events < 0 then
      failwiths "Epoll.create got nonpositive max_ready_events" max_ready_events
        ([%sexp_of: int]);
    ref (`In_use
           { epollfd = epoll_create ();
             flags_by_fd =
               Table.create
                 ~num_keys:num_file_descrs
                 ~key_to_int:File_descr.to_int
                 ~sexp_of_key:File_descr.sexp_of_t
                 ();
```

Reply:
Updated with raise an internal exn here.